server/src/main/java/org/elasticsearch/index/seqno/RetentionLeaseBackgroundSyncAction.java

```diff
@@ -90,6 +99,56 @@ public RetentionLeaseBackgroundSyncAction(
                 ThreadPool.Names.MANAGEMENT);
     }
 
+    @Override
+    protected void doExecute(Task task, Request request, ActionListener<ReplicationResponse> listener) {
+        assert false : "use RetentionLeaseBackgroundSyncAction#backgroundSync";
+    }
+
+    final void backgroundSync(ShardId shardId, String primaryAllocationId, long primaryTerm, RetentionLeases retentionLeases) {
+        final ThreadContext threadContext = threadPool.getThreadContext();
+        try (ThreadContext.StoredContext ignore = threadContext.stashContext()) {
+            threadContext.markAsSystemContext();
```

Comment:
Similarly here.

Reply:
Thanks. Fixed in https://github.com/elastic/elasticsearch/pull/48430/commits/99ccfcbdc061b011f970364a0415b9783b8d70c7.